File: app/src/main/java/com/sdin/tourstamprally/adapter/SelectLocationAdapter.java
package com.sdin.tourstamprally.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.sdin.tourstamprally.R;

import java.util.List;

public class SelectLocationAdapter extends BaseAdapter {

    private List<String> data;
    private LayoutInflater inflater;

    public SelectLocationAdapter(Context context, List<String> data) {
        clearItem();
        this.data = data;
        inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public int getCount() {
        if (data != null)
            return data.size();

        else
            return 0;
    }

    @Override
    public Object getItem(int position) {
        return data.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        if (convertView == null){
            convertView = inflater.inflate(R.layout.spinner_title_item, parent, false);

            if (data != null){
                String text = data.get(position);
                TextView textView = convertView.findViewById(R.id.spinnerTitleText);
                textView.setText(text);
            }
        }
        return convertView;
    }

    public void clearItem(){
        if (data != null){
            data.clear();
            notifyDataSetChanged();
        }
    }

    @Override
    public View getDropDownView(int position, View convertView, ViewGroup parent) {
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.spinner_item, parent, false);

        }

        ((TextView)convertView.findViewById(R.id.spinnerText)).setText(data.get(position));

        return convertView;
    }
}

File: app/src/main/java/com/sdin/tourstamprally/adapter/tour/SelectLocationAdapter.kt
package com.sdin.tourstamprally.adapter.tour

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageButton
import android.widget.TextView
import com.sdin.tourstamprally.R

class SelectLocationAdapter(val data : MutableList<String>, val context : Context) : BaseAdapter(){

    private val inflater : LayoutInflater

    init {
        clearItem()
        inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    override fun getCount(): Int {
        return data.size
    }

    override fun getItem(position: Int): Any {
        return data[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        var view = convertView
        if (view == null) {
            view = inflater.inflate(R.layout.spinner_title_item, parent, false)

            if (data != null) {
                val text = data[position]
                val textView = convertView?.findViewById<TextView>(R.id.spinnerTitleText)
                textView?.text = text

                val imageButton = convertView?.findViewById<ImageButton>(R.id.imagebtn)
                imageButton?.isEnabled = false
            }
        }
        return view!!
    }

    private fun clearItem() {
        if (data != null) {
            data.clear()
            notifyDataSetChanged()
        }
    }

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup?): View {
        var view = convertView
        if (view == null) {
            view = inflater.inflate(R.layout.spinner_item, parent, false)
        }

        view?.findViewById<TextView>(R.id.spinnerText)?.text = data[position]

        return view!!
    }
}